airflow/docker_image.go

```diff
@@ -64,6 +64,8 @@ func (d *DockerImage) Build(dockerfile, buildSecretString string, buildConfig ai
 		"-f",
 		dockerfile,
 		".",
+		"--label",
```

Comment:
I would very much prefer this label to be only added if the user is connected to Software and deploying to Software if this change is only intended for Software.  Maybe we could allow passing label array to the function, and only the Software side of deploy logic sends `io.astronomer.skip.revision` as part of that, wdyt?

Reply:
I will make this improvement.

Reply:
@neel-astro we should generalize this option similar to --build-args so that we can also expand this feature for astro cli to allow passing custom labels as well, and for software we either hardcode or expand this option with default skip label passed

Reply:
I don't mind that either, just that if we are going that route, let's ensure existing functionality is intact and we refactor other existing args used in the method